.script/tests/KqlvalidationsTests/KqlValidationTests.cs

```diff
@@ -33,14 +35,49 @@ public void Validate_DetectionQueries_HaveValidKql(string detectionsYamlFileName
             string id = res["id"];
 
             //we ignore known issues (in progress)
-            if (TemplatesToSkipValidationReader.WhiteListTemplateIds.Contains(id))
+            if (TemplatesToSkipValidationReader.WhiteListTemplateIds_MalformedQueries.Contains(id))
             {
                 return;
             }
             var validationRes = _queryValidator.ValidateSyntax(queryStr);
             Assert.True(validationRes.IsValid, validationRes.IsValid ? string.Empty : $"Template Id:{id} is not valid Errors:{validationRes.Diagnostics.Select(d => d.ToString()).ToList().Aggregate((s1, s2) => s1 + "," + s2)}");
         }
-    }
 
-}
+        [Theory]
+        [ClassData(typeof(DetectionsYamlFilesTestData))]
+        public void Validate_DetectionQueries_HaveValidTemplateStructure(string detectionsYamlFileName)
```

Comment:
not the correct place IMO - or maybe rename all pipe line - this class should be only KQL validation - Template structure validation is not related - you can create another project with shared library